src/challengepaths/2015/CommunityService.ts

```diff
@@ -0,0 +1,372 @@
+  new Requirement(["Mysticality"], {})
+);
+
+export const Moxie = new Test(
+  4,
+  "Feed Conspirators",
+  statTestPredictor($stat`moxie`),
+  new Requirement(["Moxie"], {})
+);
+
+export const FamiliarWeight = new Test(
+  5,
+  "Breed More Collies",
+  () =>
+    60 - Math.floor((familiarWeight(myFamiliar()) + weightAdjustment()) / 5),
+  new Requirement(["Familiar Weight"], {})
+);
+
+export const WeaponDamage = new Test(
+  6,
+  "Reduce Gazelle Population",
+  () => {
+    const weaponPower = getPower(equippedItem($slot`weapon`));
+    const offhandPower =
+      toSlot(equippedItem($slot`off-hand`)) === $slot`weapon`
+        ? getPower(equippedItem($slot`off-hand`))
+        : 0;
+    const familiarPower =
+      toSlot(equippedItem($slot`familiar`)) === $slot`weapon`
+        ? getPower(equippedItem($slot`familiar`))
+        : 0;
+    const songDamage =
+      SongBoom.song() === "These Fists Were Made for Punchin'" ? myLevel() : 0;
+    return (
+      60 -
+      Math.floor(
+        (getModifier("Weapon Damage") -
+          0.15 * (weaponPower + offhandPower + familiarPower) -
+          songDamage) /
+          25 +
+          0.001
+      ) -
+      Math.floor(getModifier("Weapon Damage Percent") / 25 + 0.001)
+    );
+  },
+  new Requirement(["Weapon Damage", "Weapon Damage Percent"], {})
+);
+
+export const SpellDamage = new Test(
+  7,
+  "Make Sausage",
+  () => {
+    const spicyDamage = have($skill`Intrinsic Spiciness`)
+      ? Math.min(10, myLevel())
+      : 0;
+    const dragonfishDamage =
+      myFamiliar() === $familiar`Magic Dragonfish`
+        ? numericModifier(
+            $familiar`Magic Dragonfish`,
+            "Spell Damage Percent",
+            familiarWeight($familiar`Magic Dragonfish`) + weightAdjustment(),
+            $item`none`
+          )
+        : 0;
+    return (
+      60 -
+      Math.floor((getModifier("Spell Damage") - spicyDamage) / 25 + 0.001) -
+      Math.floor(
+        (getModifier("Spell Damage Percent") - dragonfishDamage) / 25 + 0.001
+      )
+    );
+  },
+  new Requirement(["Spell Damage", "Spell Damage Percent"], {})
+);
+
+export const Noncombat = new Test(
+  8,
+  "Be a Living Statue",
+  () => {
+    const noncombatRate = -1 * getModifier("Combat Rate");
+    const unsoftcappedRate =
+      noncombatRate > 25 ? 25 + (noncombatRate - 25) * 5 : noncombatRate;
+    return 60 - 3 * Math.floor(unsoftcappedRate / 5);
+  },
+  new Requirement(["-combat"], {})
+);
+
+export const BoozeDrop = new Test(
+  9,
+  "Make Margaritas",
+  () => {
+    const familiarItemDrop = numericModifier(
+      myFamiliar(),
+      "Item Drop",
+      familiarWeight(myFamiliar()) + weightAdjustment(),
+      equippedItem($slot`familiar`)
+    );
+    if (
+      haveEquipped($item`broken champagne bottle`) &&
+      get("garbageChampagneCharge") > 0
+    ) {
+      return (
+        60 -
+        Math.floor((getModifier("Item Drop") - familiarItemDrop) / 60 + 0.001) -
```

Comment:
Might be simpler to just add a champagne factor that you assign a 1 or 2 depending on the above criteria.
```
const cFactor = haveEquipped($item`broken champagne bottle`) && get("garbageChampagneCharge") > 0 ? 2 : 1;
Math.floor((getModifier("Item Drop") - familiarItemDrop) / (cFactor * 30) + 0.001)
```